consensus/tendermint/adapter/store.go

```
@@ -231,6 +272,28 @@ func (s *Store) MakeBlock(
 	header.TimeMs = timestampMs
 	header.LastCommitHash = commit.Hash()
 
+	if height%s.config.Epoch == 0 {
+		epochId := height / s.config.Epoch
+		remoteChainNumber := uint64(0)
+		hash := common.Hash{}
+
+		header.NextValidators, header.NextValidatorPowers, remoteChainNumber, hash, err = s.gov.NextValidatorsAndPowers(epochId, remoteChainNumber, hash)
+		if err != nil {
+			log.Crit(err.Error())
```

log.Crit may ungracefully exit the process, which should be avoided.  In fact, we should just return an error, which means the proposer will not be able to propose in this round (but one of the other validators can be proposer after timeout).  It may not sign the block at this height, but be able to collect the signatures and finalize the block.